community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/StoreUpgrader.java

```diff
@@ -333,4 +311,16 @@ public UnexpectedUpgradingStoreVersionException( String filename, String expecte
             super( String.format( MESSAGE, filename, expectedVersion, actualVersion ) );
         }
     }
+
+    public static class DatabaseNotCleanlyShutDown extends UnableToUpgradeException
```

Comment:
Would be more expected to end with `Exception` in the class name, I had to read it twice.